H2TauTau/python/proto/weights/ScaleFactor.py

```diff
@@ -185,11 +185,11 @@ def getFactorJson(self, pt, eta):
                     pt_low, pt_high = re.findall("\d+\.\d+", pt_key)
                     if pt >= float(pt_low) and pt < float(pt_high):
                         return sub_eff[pt_key][u'value']
-                else:
-                    print 'pT out of range, using last pT bin'
-                    return sub_eff[pt_key][u'value']
-        else:
-            print 'JSON-based scale factor: eta out of range, last eta_high = ', eta_high, 'eta = ', eta
+                    else:
```

Comment:
I actually don't understand why you change the indentation here. This should only be reached if the for-loop completes without finding a pT match, so the implementation looks OK to me.